### ISSUE TITLE
[Bug]: Fail to load 64 collections with few entities, and not able to free memory after that

### ISSUE THREAD

**Issue by yanliang567** [milvus-io/milvus#28343]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.3-20231107-91833381
- Deployment mode(standalone or cluster): standalone
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
```

### Current Behavior

1. Milvus fails to load 64 collections with 128 partition each
2. Memory usage does not reduce in an hour after drop all the collection, and not be able to reuse the memory
3. Milvus always waits for 10 mins before refusing to load a new created collection for not enough memory

### Expected Behavior

1. load all the collection successfully.
2. Memory usage reduce in seconds for reusing.
3. Milvus refuse to load in seconds for not enough memory

### Steps To Reproduce

```markdown
1. Deploy a milvus standalone with 2c8g
2. create a collection and build index 
3. create 128 partitions, insert 100 entities into every partition, load every partition
4. repeat step 2 and 3 for 64 times
Result: Milvus fails to load 64 collections with 128 partition each
5. drop all the collections in a loop
Result: Memory usage does not reduce in an hour after drop all the collection
6. repeat step 2 and 3 for once
Result: Milvus always waits for 10 mins before refusing to load a new created collection for not enough memory. It think 10 mins is too much for predicting memory usage.
```

### Milvus Log

my pods in 4am cluster:
```
yanliang-233-etcd-0                                               1/1     Running             0              18h     10.104.23.77    4am-node27   <none>           <none>
yanliang-233-milvus-standalone-596ddcf9c-29856                    1/1     Running             0              87m     10.104.6.49     4am-node13   <none>           <none>
yanliang-233-minio-f468b8fc5-dvgqw                                1/1     Running             0              18h     10.104.23.78    4am-node27   <none>           <none>

```

**Comment by xiaofan-luan** [comment]
Set i as urgent in 2.3.4

**Comment by xiaofan-luan** [comment]
Targeting at
500 collection with 128 partition
or 10000 collection with 1 partition for 16GB standalone in 2.3.4

And it should be scaled to 10w + total partitions/collections with more resources in 2.4

**Comment by wayblink** [comment]
![image](https://github.com/milvus-io/milvus/assets/18096561/6801d188-09cf-472b-a398-38bc24b2b257)
Fail to release collection with more than 128 partitions. Etcd constraint similar to #26762

**Comment by yanliang567** [comment]
128 partitions not release issue was verified fix on 2.3-20231205-1ab538be-amd64

**Comment by yanliang567** [comment]
verified fix on milvus v2.3.4 for creating 64 collections
